pkg/datastore.go

```diff
@@ -185,30 +215,25 @@ func (s *DefaultDatastore) GetDeliveredBatch(ctx context.Context, queries []Quer
 }
 
 func (s *DefaultDatastore) GetHeaderBatch(ctx context.Context, queries []Query) ([]HeaderAndTrace, error) {
-	keys := make([]ds.Key, 0, len(queries))
+	var batch []HeaderAndTrace
```

Comment:
Same as above.  Let's avoid that alloc and keep this thing faaaaaaaast.

Reply:
How can we do that in this case exactly?

Reply:
I misread.  Please disregard